docs: add eureka blog (#938)
iresty/iresty.github.io: website/blog/2022/03/05/apisix-integration-eureka-service-discovery.md

---

> title: "How to Integrate API Gateway and Eureka?"

Reviewer: ```suggestion
title: "How to Integrate API Gateway with Eureka?"
```

authors:
  - name: "Yong Qian"
    title: "Author"
    url: "https://github.com/nic-6443"
    image_url: "https://avatars.githubusercontent.com/u/22141303?v=4"
  - name: "Fei Han"
    title: "Technical Writer"
    url: "https://github.com/hf400159"
    image_url: "https://avatars.githubusercontent.com/u/97138894?v=4"
keywords: 
- API Gateway
- Apache APISIX
- Eureka
- Service Discovery
- Servici Register

> description: This article describes how to enable Eureka as a service discovery in the API gateway Apache APISIX and how to use diagnostic tools to quickly discover links that forward exceptions.

Reviewer 2: ```suggestion
description: This article describes how to enable Eureka as a service discovery in the API gateway Apache APISIX and how to use diagnostic tools to find problems in the link.
```

tags: [Technology,Ecosystem,Service Discovery]
---

> This article describes how to enable Eureka as a service discovery in the API gateway Apache APISIX and how to use diagnostic tools to quickly discover links that forward exceptions.

<!--truncate-->
In microservices architecture, large and complex systems are vertically divided into smaller subsystems based on function or business requirements, which exist as independently deployed sub-processes that communicate with each other through network calls. How these independently deployed services discover each other is the first problem to be solved, so there is often a centralized registry in microservice architectures.

As the most core development framework in the Java ecosystem, Spring continues to liberate the productivity of Java developers from Spring MVC to Spring Boot, and Spring Cloud is Spring's answer to the micro-service architecture in the Cloud native era.

> In Spring Cloud, Eureka acts as a registry. Eureka is an open source Registry service written in The Java language by Netflix that plays a key role in Netflix's infrastructure.

Reviewer: ```suggestion
In Spring Cloud, Eureka acts as a registry. Eureka is an open source Registry service written in the Java language by Netflix that plays a key role in Netflix's infrastructure.
```

Apache APISIX is a dynamic, real-time, high-performance API gateway that provides rich traffic management features such as load balancing, dynamic upstream, canary release, circuit breaking, authentication, observability, and more. As an industry-leading microservice gateway, Apache APISIX provides native support for Eureka. This article will use the Spring Cloud demo project as an example to show you the main functions and features of Apache APISIX docking Eureka service discovery.

## Preparation Phase

This demonstration uses the official [`spring-cloud-netflix`](https://spring.io/projects/spring-cloud-netflix#overview) tutorial provided by Spring as an example, which provides the Eureka Server started with SpringBoot as the registration center of Spring Cloud. We also use the same method to start the Eureka server for demonstration. Please visit  [`spring-cloud-samples/eureka`](https://github.com/spring-cloud-samples/eureka，) for the project address.

The following will introduce you to the relevant code and startup method.

> ### Eureka Server

Reviewer 3: ```suggestion
## Eureka Server
```

> Spring Cloud provides an EnableEurekaServer annotation for Eureka, which can directly start an Eureka Server in the way of Spring Boot.

Reviewer 2: ```suggestion
Spring Cloud provides an `EnableEurekaServer` annotation for Eureka, which can directly start an Eureka Server in the way of Spring Boot.
```
